pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -2020,7 +2015,7 @@ protected abstract BrokerResponseNative processBrokerRequest(long requestId, Bro
       @Nullable Map<ServerInstance, ServerRouteInfo> offlineRoutingTable,
       @Nullable BrokerRequest realtimeBrokerRequest,
       @Nullable Map<ServerInstance, ServerRouteInfo> realtimeRoutingTable, long timeoutMs,
-      ServerStats serverStats, RequestContext requestContext)
+      ServerStats serverStats, RequestContext requestContext, @Nullable Runnable runningHandler)
```

Comment:
We need to document this new parameter. Otherwise, we will forget its actual meaning in a week. The name is also not very deterministic. `runningHandler` may mean anything. Reading the code,, it looks to me that it is a post-send hook.

Reply:
BTW, these callback make it more difficult to follow the code flow. They are fine if they are the only option, but in this case... couldn't we create an abstract `sendRequest` method, then always call `onQueryRunning` (maybe renaming it to onQuerySubmitted) and then call a reduce method? That would make the API easier to understand